core/src/amp_subgraph/metrics.rs

```diff
@@ -25,12 +34,227 @@ impl Metrics {
             "amp-process",
-            metrics_registry,
+            metrics_registry.cheap_clone(),
             store.shard().to_string(),
         );
 
-        Self { stopwatch }
+        let const_labels = [("deployment", &deployment)];
+
+        Self {
+            deployment_status: DeploymentStatus::new(&metrics_registry, const_labels.clone()),
+            deployment_head: DeploymentHead::new(&metrics_registry, const_labels.clone()),
+            deployment_target: DeploymentTarget::new(&metrics_registry, const_labels.clone()),
+            deployment_synced: DeploymentSynced::new(&metrics_registry, const_labels.clone()),
+            indexing_duration: IndexingDuration::new(&metrics_registry, const_labels.clone()),
+            blocks_processed: BlocksProcessed::new(&metrics_registry, const_labels.clone()),
+            stopwatch,
+        }
+    }
+}
+
+/// Reports the current indexing status of a deployment.
+pub(super) struct DeploymentStatus(IntGauge);
+
+impl DeploymentStatus {
+    const STATUS_STARTING: i64 = 1;
+    const STATUS_RUNNING: i64 = 2;
+    const STATUS_STOPPED: i64 = 3;
+    const STATUS_FAILED: i64 = 4;
+
+    fn new(
+        metrics_registry: &MetricsRegistry,
+        const_labels: impl IntoIterator<Item = (impl ToString, impl ToString)>,
+    ) -> Self {
+        let int_gauge = metrics_registry
+            .new_int_gauge(
+                "amp_deployment_status",
+                indoc!(
+                    "
+                    Indicates the current indexing status of a deployment.
+                    Possible values:
+                    1 - graph-node is preparing to start indexing;
+                    2 - deployment is being indexed;
+                    3 - indexing is stopped by request;
+                    4 - indexing failed;
+                    "
+                ),
+                const_labels,
+            )
+            .expect("failed to register `amp_deployment_status` gauge");
+
+        Self(int_gauge)
+    }
+
+    /// Records that the graph-node is preparing to start indexing.
+    pub fn starting(&self) {
+        self.0.set(Self::STATUS_STARTING);
+    }
+
+    /// Records that the deployment is being indexed.
+    pub fn running(&self) {
+        self.0.set(Self::STATUS_RUNNING);
+    }
+
+    /// Records that the indexing stopped by request.
+    pub fn stopped(&self) {
+        self.0.set(Self::STATUS_STOPPED);
+    }
+
+    /// Records that the indexing failed.
+    pub fn failed(&self) {
+        self.0.set(Self::STATUS_FAILED);
+    }
+}
+
+/// Tracks the most recent block number processed by a deployment.
+pub(super) struct DeploymentHead(IntGauge);
+
+impl DeploymentHead {
+    fn new(
+        metrics_registry: &MetricsRegistry,
+        const_labels: impl IntoIterator<Item = (impl ToString, impl ToString)>,
+    ) -> Self {
+        let int_gauge = metrics_registry
+            .new_int_gauge(
+                "amp_deployment_head",
```

Comment:
I think prefixing all these with `amp` when they correspond to an existing metric like `deployment_head` makes using them in dashboards much harder. When metrics mean the same thing regardless of subgraph flavor, we should use the same metric name